assets/src/dashboard/app/views/savedTemplates/content/index.js

```diff
@@ -58,27 +58,25 @@ function Content({
     'enableInProgressStoryActions'
   );
 
-  const handleMenuItemSelected = useCallback(
-    (sender, template) => {
-      setContextMenuId(-1);
-
-      switch (sender.value) {
-        case SAVED_TEMPLATE_CONTEXT_MENU_ACTIONS.OPEN_IN_EDITOR:
-          trackEvent('use_saved_template', {
-            name: template.title,
-            template_id: template.id,
-          });
-          actions.createStoryFromTemplate(template);
-          break;
-
-        case SAVED_TEMPLATE_CONTEXT_MENU_ACTIONS.PREVIEW:
-          actions.previewTemplate(null, template);
-          break;
```

Comment:
I'm not sure it's necessary to split these up.

The reason I split these up is because each `item` in the context menu takes its own `onClick` event handler now (rather than one big one that handles each item individually).

I split them up to follow a good separation of concerns, but we still need to divvy them up to each menu item. That's still split up by the `value` of the item. Ex:
```
menuItemActions: {
        default: () => setContextMenuId(-1),
        [STORY_CONTEXT_MENU_ACTIONS.COPY_STORY_LINK]: handleCopyStoryLink,
        [STORY_CONTEXT_MENU_ACTIONS.CREATE_TEMPLATE]: handleCreateTemplateFromStory,
        [STORY_CONTEXT_MENU_ACTIONS.DELETE]: handleDeleteStory,
        [STORY_CONTEXT_MENU_ACTIONS.DUPLICATE]: handleDuplicateStory,
        [STORY_CONTEXT_MENU_ACTIONS.OPEN_STORY_LINK]: handleOpenStoryInEditor,
        [STORY_CONTEXT_MENU_ACTIONS.RENAME]: handleRenameStory,
},```

Do y'all have any better ideas? Maybe we should keep the old way that feels like a reducer pattern.

Reply:
moved to #6516